ledger/eval.go

```diff
@@ -463,6 +468,10 @@ func startEvaluator(l ledgerForEvaluator, hdr bookkeeping.BlockHeader, proto con
 			eval.block.BlockHeader.GenesisHash = eval.genesisHash
 		}
 		eval.block.BlockHeader.RewardsState = eval.prevHeader.NextRewardsState(hdr.Round, proto, incentivePoolData.MicroAlgos, prevTotals.RewardUnits())
+		eval.specials = transactions.SpecialAddresses{
+			FeeSink:     eval.block.BlockHeader.FeeSink,
+			RewardsPool: eval.block.BlockHeader.RewardsPool,
+		}
```

Comment:
startEvaluator() can be invoked with a block header that does not have them set up, so the first setting leaves them as the zero address in that case.  Then, after calling NextRewardsState, they are set up and can be set (again, but this time properly).  I could not convince myself that it was ok to remove the initial setting, since this second setting only happens conditionally.

I'll look a little closer to see if I can confirm this makes sense, or if the startEvaluator() calls should always have a block hdr with the special addrs set.

Reply:
Ok, improved. We were not filling in the RewardsState when creating a "next header" (because much of it needs to be calculated). Therefore it was empty at the start. But we need the special addresses to do evaluation.  So now, `bookkeeping.MakeBlock` carries forward the special addresses, even though it can't fill the entire RewardsState.  Therefore, only the top assignment in startEvaluator() is required.